src/edu/stanford/nlp/sempre/ExecutorCache.java

```diff
@@ -0,0 +1,70 @@
+package edu.stanford.nlp.sempre;
+
+import java.util.*;
+
+import fig.basic.*;
+
+/**
+ * Cache the executed values of an executor.
+ *
+ * The cache should be cleared once the parser finishes each example.
+ * The metakey Object specifies the current example.
+ * When the metakey is changed, the cache is cleared.
+ *
+ * @author ppasupat
+ */
+public final class ExecutorCache {
```

Comment:
Actually the class is not used in the core module. (It's specific to my LambdaDCS executor.) I will move it to my tables package and will submit it in another pull request.

My LambdaDCS executor handles parallel processing by having one executor cache per example.